### PR TITLE
feat: add prometheus metrics endpoint

### PR DESCRIPTION
**Pull request by wa0x6e**
## 🧿 Current issues / What's wrong ?

From https://github.com/snapshot-labs/pitches/issues/18

## 💊 Fixes / Solution

This PR add a `/metrics` endpoint, exposing some basic express/node metrics, as well as custom metrics

## 🚧 Changes

- Expose metrics via `/metrics` endpoint

## 🛠️ Tests

- Go to `http://localhost:3006/metrics`

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -27,6 +31,7 @@ app.use(
   )
 );
 app.use(cors({ maxAge: 86400 }));
+app.use(favicon(path.join(__dirname, '../public', 'favicon.ico')));
```
Comment (Todmy) [inline src/index.ts:34]: can we use this approach without any extra libraries?
```suggestion
app.get('/favicon.ico', (req, res) => res.status(204)); 
```

Comment (wa0x6e) [reply src/index.ts:34]: What you're suggesting will not serve favicon at all

Comment (Todmy) [reply src/index.ts:34]: yes, why do you need it for API?

Comment (wa0x6e) [reply src/index.ts:34]: There's the `/preview` endpoint, which is not an API.

And without this, polling the `/metrics` endpoint will log the favicon, and increase counters

Comment (wa0x6e) [reply src/index.ts]: To test: 
- comment the favicon line in index.ts
- visit /metrics, and hard refresh the page
- you will see a counter for the favicon request
- refresh the page
- the favicon counter increase
- => this is undesired, as each call to the /metric endpoint will increase the GET counter

Now, without commenting the favicon line, doing the same steps above, there is no favicon counters in the metrics page

**Review by Todmy — APPROVED** [review]
utACK

**Pull request merged**